controllers/volumereplication_controller.go

```diff
@@ -196,14 +196,39 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		replicationErr = r.markVolumeAsPrimary(instance, logger, volumeHandle, parameters, secret)
 
 	case replicationv1alpha1.Secondary:
-		replicationErr = r.markVolumeAsSecondary(instance, logger, volumeHandle, parameters, secret)
-		// resync volume if successfully marked Secondary
-		if replicationErr == nil {
-			err := r.updateReplicationStatus(instance, logger, getReplicationState(instance), "volume is marked secondary")
-			if err != nil {
-				return ctrl.Result{}, err
+		// For the first time, Mark the volume as secondary and requeue the
+		// request, For some storage providers it takes some time to determine
+		// whether the volume need correction example:- correcting split brain.
+		if instance.Status.State != replicationv1alpha1.SecondaryState {
```

Comment:
`instance.Status.State` would be `nil` for the first reconcile. And  if the user creates the volume replication resource with replication state as `Secondary`, this if condition might fail. So IMO, we should check `instance.Status` not nil as well. What do you think?

Reply:
`instance.Status` is not a pointer I think we don't need nil check here, i didn't saw any error when CR is created with secondary state